### PR TITLE
[W5.3b][W14-A2]Justin Poh

### PR DESCRIPTION
**Pull request by justinpoh**
Extracted commonalities from Phone, Email and Address classes into a parent class called Contact

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -6,27 +6,23 @@
...
-    public final String value;
-    private boolean isPrivate;
-
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
+        super(address.trim(), isPrivate);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: here you are trying to assign value before validating it.

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package seedu.addressbook.data.person;
+
+/**
+ * This is the parent class for Phone, Email and Address, so as to extract
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java]: No need to talk about the child classes here. Just sufficient if you mention what this class is about.

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * as much commonality and reduce method duplication
+ */
+
+public class Contact {
+    public final String value;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java]: try to make it private. in general, start off with private members, then use protected or public only if absolutely necessary.

File: src/seedu/addressbook/data/person/Contact.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    @Override
+    public String toString(){
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return this == other
+                || (other instanceof Contact
+                && this.value.equals(((Contact) other).value));
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java]: This can potentially equate a phone number with address, think why?

**Comment by okkhoy** [comment]
@justinpoh 
Some comments added. Please ack & close the PR after reading comments.

**Comment by justinpoh** [comment]
Acknowledged.

**Pull request closed without merging**